misc/systemd/kubo-hardened.service

```diff
@@ -1,29 +1,29 @@
 # This file will be overwritten on package upgrades, avoid customizations here.
 #
 # To make persistent changes, create file in 
-# "/etc/systemd/system/ipfs.service.d/overwrite.conf" with 
-# `systemctl edit ipfs.service`. This file will be parsed after this 
+# "/etc/systemd/system/kubo.service.d/overwrite.conf" with
+# `systemctl edit kubo.service`. This file will be parsed after this
 # file has been parsed.
 #
 # To overwrite a variable, like ExecStart you have to specify it once
 # blank and a second time with a new value, like:
 # ExecStart=
-# ExecStart=/usr/bin/ipfs daemon --flag1 --flag2
+# ExecStart=/usr/bin/kubo daemon --flag1 --flag2
 #
 # For more info about custom unit files see systemd.unit(5).
 
-# This service file enables systemd-hardening features compatible with IPFS,
+# This service file enables systemd-hardening features compatible with Kubo,
 # while breaking compatibility with the fuse-mount function. Use this one only 
 # if you don't need the fuse-mount functionality.
 
 [Unit]
-Description=InterPlanetary File System (IPFS) daemon
+Description=Kubo daemon for the InterPlanetary File System (IPFS)
 Documentation=https://docs.ipfs.io/
 After=network.target
 
 [Service]
 # hardening
-ReadWritePaths="/var/lib/ipfs/"
+ReadWritePaths="/var/lib/kubo/"
```

Comment:
What is stored in this directory? 
Are we breaking wexisting users?

Reply:
That's the default data path for services. It's up for the package maintainers to set this up as the user directory for the user kubo or previously ipfs.

The users in this case are the package maintainers, which should see the change in the file and adjust their install script accordingly.

I move the data from the old to the new directory like this:

```bash
post_upgrade() {
  
  ## search for old location of kubo data and move data

  if [ -f '/var/lib/ipfs/config' ]; then
    daemon_was_running='no'
    echo ":: ipfs-to-kubo upgrade: stopping ipfs.service if running..."
    if [ $(systemctl is-active --quiet ipfs.service) ]; then
      daemon_was_running='yes'
      systemctl stop ipfs.service >/dev/null 2>&1 || true
    fi
    echo ":: moving existing ipfs data from '/var/lib/ipfs/' to '/var/lib/kubo/'..."
    mkdir -p /var/lib/kubo
    mv /var/lib/ipfs /var/lib/kubo >/dev/null 2>&1 || true
    chown kubo: -R /var/lib/kubo/ || true
  fi

  # in case the daemon was never started, generate a configuration

  if [ ! -f /var/lib/kubo/config ]; then
    systemctl restart systemd-sysusers.service

    mkdir -p /var/lib/kubo/
    chown kubo: -R /var/lib/kubo/
    su kubo -s /bin/bash -c "export IPFS_PATH='/var/lib/kubo/'; kubo init"
  else
    if [ $(systemctl is-active --quiet kubo.service) ]; then
      daemon_was_running='yes'
      systemctl stop kubo.service >/dev/null 2>&1 || true
    fi
  fi

  # do data migration
  
  su kubo -s /bin/bash -c "export IPFS_PATH='/var/lib/kubo/'; kubo repo migrate"
  
  if [ "$daemon_was_running" == 'yes' ]; then
    systemctl start kubo.service >/dev/null 2>&1 || true
  fi

}
```

Reply:
> I am concerned about the side effects here: as noted in https://github.com/ipfs/kubo/issues/8959 the rule of thumb is to keep changes minimal, avoiding breaking existing users and tutorials that were created in the past 7 years.
> 
> Changing binary name, user ang group sound like extremely breaking change, no?
> How things will behave for existing users?

Well at least the hardened version of the service file is not super critical, as it's probably not used in *that* many places. I've created it fairly recently, and I don't think we document it anywhere (yet).